src/components/sbb-datepicker/sbb-datepicker.tsx

```diff
@@ -210,31 +191,34 @@ export class SbbDatepicker implements ComponentInterface {
   }
 
   private async _valueChanged(event): Promise<void> {
-    await this._formatAndUpdateValue(event.target.value);
+    await this._formatAndUpdateValue(event.target.value, this._parse(event.target.value));
   }
 
   /** Applies the correct format to values and triggers event dispatch. */
-  private async _formatAndUpdateValue(value: string): Promise<void> {
+  private _formatAndUpdateValue(value: string, valueAsDate: Date): void {
     if (this._inputElement) {
-      const newValueAsDate = this._parse(value);
-      this._inputElement.value = this._formatValue(value, newValueAsDate);
+      this._inputElement.value = !this._dateAdapter.isValid(valueAsDate)
+        ? value
+        : this._format(valueAsDate);
 
       const isValidOrEmpty =
         !value ||
-        (newValueAsDate &&
+        (valueAsDate &&
           isDateAvailable(
-            newValueAsDate,
+            valueAsDate,
             this._element.dateFilter,
             this._inputElement?.min,
             this._inputElement?.max,
           ));
       toggleDatasetEntry(this._inputElement, 'sbbInvalid', !isValidOrEmpty);
-      this._emitChange();
+      this._emitChange(valueAsDate);
```

Comment:
Is this the only place we call _emitChange() ?

Reply:
Yes it is